### ISSUE TITLE
Popover

### ISSUE THREAD

**Issue by debrice** [StartTheShift/shift-components#10]
- Custom trigger: focus, click and hover. 
- Loads template string or URL
- optional controller ?
- Positioning should probably be relative to the element it is attached since absolute positioning breaks on scroll
 - We should be able to get the target from the event, then appendchild a size less element, with a relative positioning as a popover container.

**Comment by jimobrien** [comment]
Probably will need to be a `restrict: 'A'` directive so that any element can be the anchor for a popover 

e.g.
```jade
a(shift-popover) sometext

div(shift-popover) sometext

ul
  li(shift-popover) sometext
```

ect.

**Comment by debrice** [comment]
I concur, no need for it to be an element directive, makes sense to restrict it to be an attribute. 

What do you think about delegating the rendering to a sub directive? The popover directive would handle events and prevent simultaneous popover.

**Comment by debrice** [comment]
Also using an attribute directive might actually preventing us from using I a transclusion tho... I was kind of looking forward transcluding the popover content ^^

**Comment by jimobrien** [comment]
:+1: to rending with a sub directive.

No worries, transclusion will still work with an attribute.

**Comment by debrice** [comment]
I was thinking of that sort of case:

```jade
button.button(
  popover = "hover"
  ng-click = "doStuff()"
) Click Me
```

Transcluding inside that button would then compromise the button's caption

**Comment by jimobrien** [comment]
I think the popover would just need to be a child in that case:
```jade
button.button(
  ng-click = "doStuff()"
) 
  div(shift-popover)
  | Click Me
```

**Comment by debrice** [comment]
Brainstorm, what about that:

```jade
button.button(
  ng-click = "doStuff()"
  shift-popover = "hover"
) click me
  shift-popover-content
    p
      |  I am the popover content

```

**Comment by debrice** [comment]
actually a popover could automatically find its parent element and add an event listener on it.

**Comment by jimobrien** [comment]
Hm. I feel like that begins to destruct the encapsulated nature of an isolated component. Now what could be an internal of `shift-popover` is spread out across multiple directives.

**Comment by debrice** [comment]
```jade
button.button(
  ng-click = "doStuff()"
) click me
  shift-popover(
    on = "hover"
  )
    p
      |  I am the popover content
```

**Comment by jimobrien** [comment]
ooOo I like that

**Comment by debrice** [comment]
So we can have 
```jade
p
  | this is some text with a 
  span technical word
    shift-popover(on="hover") Word definition
 | and the text continues here...
    
```

**Comment by debrice** [comment]
I don't think we should provide a mean to include a template URL. For complex popover, like form in a popover, I'd recommend the form being in its own directive.
